File: src/main/java/ru/market/pricing_service/model/dto/PriceTag.java
package ru.market.pricing_service.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PriceTag {
    private String productName;
    private BigDecimal price;
    private String tagType; // белый, желтый, акционный
    private BigDecimal regularPrice;
    private String promoDescription = null; // Для поддержки описания специальных акций типа "1+1", "2+1" и т.д.
}

File: src/main/java/ru/market/pricing_service/model/dto/PriceTag.kt
package ru.market.pricing_service.model.dto

import java.math.BigDecimal

data class PriceTag(
    val productName: String,
    val price: BigDecimal,
    val tagType: String, // белый, желтый, акционный
    val regularPrice: BigDecimal,
    val promoDescription: String? = null // Добавлено для поддержки описания специальных акций типа "1+1", "2+1" и т.д.
)